src/validation.h

```diff
@@ -994,6 +994,16 @@ class ChainstateManager
      */
     bool ProcessNewBlockHeaders(const std::vector<CBlockHeader>& block, BlockValidationState& state, const CChainParams& chainparams, const CBlockIndex** ppindex = nullptr) LOCKS_EXCLUDED(cs_main);
 
+    /**
+     * Try to add a transaction to the memory pool.
+     *
+     * @param[in]  tx              The transaction to submit for mempool acceptance.
+     * @param[in]  bypass_limits   When true, don't enforce mempool fee and capacity limits.
+     * @param[in]  test_accept     When true, run validation checks but don't submit to mempool.
+     */
+    [[nodiscard]] MempoolAcceptResult ProcessTransaction(const CTransactionRef& tx, bool bypass_limits, bool test_accept=false)
```

Comment:
In commit "[validation] Add CChainState::ProcessTransaction()" (bbc785f31b5d70c702ae37d259c46652e243ddba)

IMO it would be really good to add this function in its final form without the `bypass_limits` parameter, so the later commit removing it can be dropped. This would simplify reviewing the PR, and more importantly avoid unsafe change:

```diff
-    [[nodiscard]] MempoolAcceptResult ProcessTransaction(const CTransactionRef& tx, bool bypass_limits, bool test_accept=false)
+    [[nodiscard]] MempoolAcceptResult ProcessTransaction(const CTransactionRef& tx, bool test_accept=false)
```

Change is unsafe because a `ProcessTransaction(tx, true)` call would still compile and silently change meaning from "bypass limits and don't test" to "don't bypass limits and do test"

Better to avoid this churn and risk, and just add the new method in final form without temporary (bool, bool) params.

Reply:
Good idea. This was a misguided attempt to make the commits as granular as possible, but you're right that it just adds churn and confusion.